docs/src/pages/customization/components/DynamicThemeNesting.tsx

```diff
@@ -1,19 +1,31 @@
 import React from 'react';
 import Button from '@material-ui/core/Button';
-import { createMuiTheme, ThemeProvider } from '@material-ui/core/styles';
+import { createMuiTheme, ThemeProvider, Theme } from '@material-ui/core/styles';
 import FormControlLabel from '@material-ui/core/FormControlLabel';
 import { blue } from '@material-ui/core/colors';
 import Switch from '@material-ui/core/Switch';
 
-const defaultTheme = createMuiTheme();
-
 export default function DynamicThemeNesting() {
   const [color, setColor] = React.useState('default');
 
   const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
     setColor(event.target.checked ? 'blue' : 'default');
   };
 
+  const theme = React.useMemo(() => {
+    if (color === 'blue') {
+      return createMuiTheme({
+        palette: {
+          secondary: {
+            main: blue[500],
+            contrastText: '#fff',
+          },
+        },
+      });
+    }
+    return createMuiTheme();
```

Comment:
An argument can be made that this should return the outer `defaultTheme`. I opted to improve startup time of the app rather than update time (By doing work inside a component React will be able to time slice the work properly. This isn't possible if work is done during module evaluation.)

Reply:
I think that it's great like this. The outer theme might be null, like in CodeSandbox.

Reply:
I was referencing the deleted line. Basically `const moduleTheme = createMuiTheme()` which is defined outside the function body.

Reply:
👍